spec/controllers/groups_controller_spec.rb

```diff
@@ -287,13 +287,114 @@
         end
       end
     end
-    describe '#invalidate_groupings'
-    describe '#validate_groupings'
-    describe '#delete_groupings'
-    describe '#add_members'
-    describe '#add_member'
-    describe '#remove_members'
-    describe '#remove_member'
+
+    describe '#validate_groupings' do
+      let(:grouping) { create :grouping_with_inviter }
+
+      it 'should validate groupings' do
+        post :global_actions, params: {
+            assignment_id: grouping.assignment.id,
+            groupings: [grouping.id],
+            global_actions: 'valid'
+        }
+        expect(grouping.reload.admin_approved).to be true
+      end
+    end
+
+    describe '#invalidate_groupings' do
+      let(:grouping) { create :grouping_with_inviter }
+
+      it 'should invalidate groupings' do
+        post :global_actions, params: {
```

Comment:
Try not to call the method that you're testing in the setup for your test. This either duplicates your test at line 294 or it assumes that the method must work properly in order to test if the method works properly. 

Instead, please set the attributes you want to change when you create the model object using the factory. So something like:

```ruby
let(:grouping) { create :grouping_with_inviter, admin_approved: false }
``` 

and then just check that it gets swapped to `true` after you call the `global_actions` method.

The same for the other tests below.